fetch-todolist/src/component/Home.js

```diff
@@ -1,76 +1,115 @@
+import { useMutation, useQuery, useQueryClient } from "react-query";
+import { getTodos, deleteTodo } from "../services/todos.services.js";
+import "./styling.css";
 
 const Home = () => {
-  const [list, setList] = useState([]);
   const [idToDel, setIdToDel] = useState();
-  const [isBlockHidden, setIsBlockHidden] = useState(false);
   const [isDelBlockHidden, setIsDelBlockHidden] = useState(false);
-  const { loading, error, data } = useFetch();
-  useEffect(() => {
-    setList(data);
-  }, [data]);
+  const pageSize = 4;
+  const [pageNumber, setPageNumber] = useState(1);
+  const [displayTodos, setTodos] = useState([]);
+  const [order, setOrder] = useState();
+  const queryClient = useQueryClient();
+  const [search, setSearch] = useState("");
+
+  const { data, isLoading, error } = useQuery(
+    ["todos", pageNumber, pageSize, order, search],
+    () => {
+      return getTodos({
+        _page: pageNumber,
+        _limit: pageSize,
+        _sort: "task",
+        _order: order,
+        task_like: search,
+      });
+    }
+  );
 
   const changeStatus = (check, id) => {
-    const updatedStatusArray = list.map((obj) => {
+    const updatedStatusArray = data.map((obj) => {
       if (obj.id === id) {
         obj.status = check;
       }
       return obj;
     });
-    setList(updatedStatusArray);
+    setTodos(updatedStatusArray);
   };
 
+  const { mutate: deleteTodoMutation } = useMutation(deleteTodo, {
+    onSuccess: () => {
+      queryClient.invalidateQueries("todos");
+    },
+  });
+
   const handleDelete = (id) => {
-    const resltArray = list.filter((filteredItem) => id !== filteredItem.id);
-    setList(resltArray);
-    openDelDiv();
-  };
-  const openDiv = () => {
-    let condition = !isBlockHidden ? true : false;
-    setIsBlockHidden(condition);
+    deleteTodoMutation(id);
+    openDelDiv(null);
   };
+
   const openDelDiv = (id) => {
-    let condition = !isDelBlockHidden ? true : false;
-    setIsDelBlockHidden(condition);
-    setIdToDel(id);
+    setIsDelBlockHidden(!isDelBlockHidden);
+    id && setIdToDel(id);
   };
-  const handleAdd = (task, ddate) => {
-    if (task !== "" && ddate !== "") {
-      const obj = {
-        id: list.length + 1,
-        task: task,
-        dueDtae: ddate.toString(),
-        status: false,
-      };
-      setList([...list, obj]);
-      setIsBlockHidden(false);
+
+  const handleStatusFilter = (e) => {
+    let completedTask = data;
+    let status = e.target.value;
+
+    if (status === "Completed") {
+      completedTask = completedTask.filter((task) => task.status);
+    } else if (status === "Active") {
+      completedTask = completedTask.filter((task) => !task.status);
     }
+    setTodos(completedTask);
   };
 
+  const searchTask = (value) => {
+    setSearch(value);
+  };
+
+  const handleOrder = (event) => {
+    setOrder(event.target.value);
+  };
+
+  useEffect(() => {
+    setTodos(data);
+  }, [data]);
   return (
```

Comment:
Indentation is not up to mark, please use prettier

Reply:
already I'm using it, but i'll still have another look.